### ISSUE TITLE
Persist queue

### ISSUE THREAD

**Issue by rtholmes** [ubccpsc/classy#206]
The only thing that isn't persisted right now if the service is rebooted is the queues. This should actually be pretty straight-forward: on `tick` write the queues (and currently executing jobs) to disk. When the queue is started the first time, see if there's a file on disk to read.

Not sure what the implications are for this in terms of the container reading/writing to disk that can persist through the container being lost.

**Comment by nickbradley** [comment]
On classy.cs, I saw `express.json`, `regression.json`, and `standard.json` in `/var/opt/classy/runs` which I assume are for queue persistence. Not sure if this is the best location since there is an archive script that runs periodically and moves old runs to another dir. I can always mount a `/queue` directory if that is easier.

**Comment by rtholmes** [comment]
Yeah, I was just writing to the PERSIST_DIR. Where would it go if we just write to `./` instead?

**Comment by nickbradley** [comment]
Then it would only be stored temporarily inside the container. I'll mount another dedicated directory and specify it using QUEUE_DIR.

**Comment by rtholmes** [comment]
@nickbradley just checking: does this exist now? aka should I update the CI `.env` etc? If it doesn't exist, should we make it a bit more robust instead of `QUEUE_DIR` and use something more generic in case we need it for other use cases in the future (so we won't have to update all existing `.env` files), something like `CLASSY_PERSIST_DIR`?

**Comment by nickbradley** [comment]
I haven't done anything yet. I like the idea of having a more generic persist dir for classy. If you want to make the changes to the code, just set `CLASSY_PERSIST_DIR` to `PERSIST_DIR` and I'll try to update the docker config this weekend (mostly just need to update the docs to ensure that the classy persist dir actually exists before mounting).

**Comment by rtholmes** [comment]
We can just open this in a PR and I'll add my changes once you're done. There's not a huge rush on this, and we'll need to be clear when forks update that they will need a new `.env` variable.
